transmart-core-db/grails-app/services/org/transmartproject/db/tree/TreeService.groovy

```diff
@@ -196,24 +221,42 @@ class TreeService implements TreeResource {
      * @param currentUser the current user.
      * @throws ServiceNotAvailableException iff a rebuild operation is already in progress.
      */
-    @Async
-    void rebuildCache(org.transmartproject.core.users.User currentUser) throws ServiceNotAvailableException {
-        if (!cacheRebuildLock.tryLock()) {
-            throw new ServiceNotAvailableException("Rebuild operation already in progress.")
-        }
-        log.info "Clearing all caches ..."
-        def stopWatch = new StopWatch('Rebuild cache')
-        stopWatch.start("Clearing the caches")
-        clearCache(currentUser)
-        stopWatch.stop()
-        usersResource.getUsers().each { User user ->
-            log.info "Rebuilding the cache for user ${user.username} ..."
-            stopWatch.start("Rebuild the cache for ${user.username}")
-            treeCacheService.fetchCachedSubtree(user.admin, getStudyTokens(user), I2b2Secure.ROOT, 0)
-            stopWatch.stop()
-        }
-        log.info "Done rebuilding the cache.\n${stopWatch.prettyPrint()}"
-        cacheRebuildLock.unlock()
+    void rebuildCache(User currentUser) throws ServiceNotAvailableException {
```

Comment:
What was wrong with the old way? Didn't `Async` work?

Reply:
No, it didn't :(
In normal Spring Boot it works, but Grails apparently has its own methods.
Probably we could make the Spring Boot way to work, but I don't know how.